### ISSUE TITLE
Question about independent_first_frame usage

### ISSUE THREAD

**Issue by LiHaoy-ux** [thu-ml/Causal-Forcing#24]
Regarding the **frame-wise Teacher Forcing** training for the Causal I2V model: I noticed that when **independent_first_frame** is set to True in default.yaml, the implementation in **causal_model.py** raises a **NotImplementedError**. Is there a specific reason why this mode isn't supported for teacher forcing yet? Is it due to the complexity of the FlexAttention mask construction for independent prefixes?

<img width="868" height="214" alt="Image" src="https://github.com/user-attachments/assets/3a76edce-0bb0-4b89-a1f2-dc8a5d93251e" />

**Comment by zhuhz22** [comment]
It was simply not implemented in time. There should be no issue with FlexAttention, so you can go ahead and implement it.

**Comment by LiHaoy-ux** [comment]
Current training treats the first frame as a noisy latent ($z_t$), just like T2V. However, during I2V inference, we use a clean reference image ($z_0$). Does this noise distribution mismatch between training and inference affect the model's ability to preserve the first frame's identity?
 I attempted to implement `independent_first_frame: true` by treating the first frame as a clean, fully-attended global prefix (non-symmetric mask). However, even on an **H20 (90GB)**, this triggered severe **OutOfMemoryErrors (OOM)** during the `optimizer.step()`.It seems that enforcing a global attention prefix for all subsequent frames significantly increases the activation memory peak, especially in a teacher-forcing setup where we handle dual sequences (clean & noisy). How can I solve it？

**Comment by Anachronism-N** [comment]
How many steps are recommended for the DMD training in stage 3 of the chunkwise i2v model? I found that the quality of the model's generation would significantly decline when it exceeded 200 steps. From the second block, there would be a large amount of noise, and the noise would become more and more severe as it progressed.

**Comment by zhuhz22** [comment]
> How many steps are recommended for the DMD training in stage 3 of the chunkwise i2v model? I found that the quality of the model's generation would significantly decline when it exceeded 200 steps. From the second block, there would be a large amount of noise, and the noise would become more and more severe as it progressed.

Hi @Anachronism-N , it's quite strange that `From the second block, there would be a large amount of noise`, and I think there may be bugs in your implementation. You can join our WeChat group for detailed and convenient discussion.
